iota-client/src/builder.rs

```diff
@@ -169,7 +184,7 @@ impl ClientBuilder {
             mqtt_topic_handlers: Default::default(),
             #[cfg(feature = "mqtt")]
             broker_options: self.broker_options,
-            local_pow: self.local_pow,
+            network_info: network_info.read().unwrap().clone(),
```

Comment:
you're updating the `network_info` on the Client::sync_nodes function but it won't reflect the changes on the field here; you should store the Arc<RwLock<NetworkInfo>> instead

Reply:
Thanks, updated